### PR TITLE
Breakdown Background.js

### PR DESCRIPTION
**Pull request by zoracon**
### First attempt breaking down this file to accomplish:
1. Writing in more documentation via jsDoc
2. Making the code more readable
3. Removing utility code into modules
4. Making the code easier to migrate to potential MV3 port (#17268)

Doing some extensive testing before merge of course, I expect I may fail some linting and plan to prune that soon. My directive this week is to have this up by the end of the work week so I can start setting up an issue for the next release.

Edit: Decided to go a step further and breakdown this file more, won't be pushing in this upcoming release, but will be in the next one for particular reasons.

### CODE REVIEW

File: chromium/background-scripts/background.js [hunk 1]
```diff
@@ -559,37 +486,18 @@ function onCompleted(details) {
...
-      details.error.indexOf("The server uses key pinning (HPKP) but no trusted certificate chain could be constructed that matches the pinset. Key pinning violations cannot be overridden.") == 0 ||
-      details.error.indexOf("SSL received a weak ephemeral Diffie-Hellman key in Server Key Exchange handshake message.") == 0 ||
-      details.error.indexOf("The certificate was signed using a signature algorithm that is disabled because it is not secure.") == 0 ||
-      details.error.indexOf("Unable to communicate securely with peer: requested domain name does not match the server’s certificate.") == 0 ||
-      details.error.indexOf("Cannot communicate securely with peer: no common encryption algorithm(s).") == 0 ||
-      details.error.indexOf("SSL peer has no certificate for the requested DNS name.") == 0
-    )) {
+    session.getRequest(details.requestId, "simple_http_nowhere_redirect", false) &&
+    // Enumerate errors that are likely due to HTTPS misconfigurations
+    ssl_codes.error_list.includes(details.error)
```
Comment (cschanaj) [inline chromium/background-scripts/background.js]: I believe in the current implementation, we are looking for partial string match for each message instead, see [background.js#L564](https://github.com/EFForg/https-everywhere/blob/872e5ced15bed896c9c05f3e63ea10d8be4ef757/chromium/background-scripts/background.js#L564)

```suggestion
    ssl_codes.some(message => details.error.indexOf(message) == 0)
```

File: chromium/background-scripts/background.js [hunk 2]
```diff
@@ -488,7 +488,7 @@ function onErrorOccurred(details) {
     details.type == "main_frame" &&
     session.getRequest(details.requestId, "simple_http_nowhere_redirect", false) &&
     // Enumerate errors that are likely due to HTTPS misconfigurations
-    ssl_codes.error_list.includes(details.error)
+    ssl_codes.some(message => details.error.indexOf(message) == 0)
```
Comment (pipboy96) [inline chromium/background-scripts/background.js]:
```suggestion
    ssl_codes.some(message => details.error.includes(message))
```

File: chromium/manifest.json [hunk 3]
```diff
@@ -9,7 +9,12 @@
     "background": {
         "scripts": [
             "background-scripts/bootstrap.js",
-            "background-scripts/util.js",
+						"background-scripts/log.js",
+						"background-scripts/validation.js",
+						"background-scripts/modules/ip_utils.js",
+						"background-scripts/modules/browser_session.js",
```
Comment (cschanaj) [inline chromium/manifest.json]: It seems that the indents here are off

Comment (zoracon) [reply chromium/manifest.json]: My code editor strikes again. I'll fix that.

**Comment by zoracon** [comment]
Update: Waiting until major work is done on Tor side.

**Comment by zoracon** [comment]
Closing this PR and pushing an updated PR that reflects the same changes. I just had to restructure the refactor branch to reflect "greenfield" status for this restructure.

**Pull request closed without merging**